### PR TITLE
feat(ecl-editor): add editor CSS rules - INNO-1846

### PR DESCRIPTION
**Pull request by emeryro**
# PR description

The first step here is to decide how we will handle the editor CSS.
Here is the RFC: https://github.com/ec-europa/europa-component-library/blob/feat/ecl-editor-INNO-1846/docs/decisions/003-editor.md

Playground: https://5dce12d52570804233ab3ecb--europa-component-library.netlify.com/playground/ec/?path=/story/utilities-editor--sample

### CODE REVIEW

**Comment by HAL-Patch-INNO** [comment]
Deploy preview for *europa-component-library* ready!

Built with commit a14206c1a024fa7c267cf9c1abb7483c0580f1db

https://deploy-preview-1360--europa-component-library.netlify.com

**Comment by yhuard** [comment]
My 2¢:
- Mixed CSS rules: 👎 (will break our BEM conventions linting process)
- Editor preset, with duplicated rules 👍👍 
  - "It may be tricky for developpers to use this preset specifically when needed (to be analysed)": if the developper has a doubt, she can embed the stylesheet on all the pages. It means 1 extra network request but it's not that bad.
- Editor preset, with SASS mixins 😕 
  - another con: it makes the code much less readable...
  - also: it might be hard to handle cases where the markup is slightly different in a wysiwyg compared to the markup of the component we provide
- Separated package for editors 👍
  - IMHO it makes it easier to develop and maintain

File: docs/decisions/003-editor.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
+# Handle `editor` CSS rules in ECL
+
+| Status        | proposed                     |
```
Comment (yhuard) [inline docs/decisions/003-editor.md]:
```suggestion
| Status        | accepted                     |
```

File: docs/decisions/003-editor.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+**Cons**
+
+- Will increase size of components
+- Not really BEM compliant
+
+**Example**
+
+ec-component-table.scss:
+
+```
```
Comment (yhuard) [inline docs/decisions/003-editor.md]:
```suggestion
```scss
```

File: docs/decisions/003-editor.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+**Cons**
+
+- Yet another preset to handle
+- It may be tricky for developpers to use this preset specifically when needed (to be analysed)
+
+**Example**
+
+ec-component-table.scss:
+
+```
```
Comment (yhuard) [inline docs/decisions/003-editor.md]:
```suggestion
```scss
```

File: docs/decisions/003-editor.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+
+.ecl-table__head {
+  display: none;
+}
+...
+```
+
+ec-component-table-editor.scss:
+
+```
```
Comment (yhuard) [inline docs/decisions/003-editor.md]:
```suggestion
```scss
```

File: docs/decisions/003-editor.md [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+
+**Cons**
+
+- Yet another preset to handle
+- It may be tricky for developpers to use this preset specifically when needed (to be analysed)
+- Would require additional work to update existing component and split them into mixins
+
+mixin.scss:
+
+```
```
Comment (yhuard) [inline docs/decisions/003-editor.md]:
```suggestion
```scss
```

File: docs/decisions/003-editor.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+}
+
+.ecl-table__head {
+  @include ecl-table-head();
+}
+```
+
+ec-component-table-editor.scss:
+
+```
```
Comment (yhuard) [inline docs/decisions/003-editor.md]:
```suggestion
```scss
```

File: docs/decisions/003-editor.md [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+}
+
+@mixin ecl-table-head() {
+  display: none;
+}
+```
+
+ec-component-table.scss:
+
+```
```
Comment (yhuard) [inline docs/decisions/003-editor.md]:
```suggestion
```scss
```

**Review by yhuard — CHANGES_REQUESTED** [review]
now what would be nice is to create a dummy page in storybook where we can see it in action ;)

**Comment by emeryro** [comment]
playground added: https://5dce12d52570804233ab3ecb--europa-component-library.netlify.com/playground/ec/?path=/story/utilities-editor--sample

**Review by yhuard — CHANGES_REQUESTED** [review]
- The table looks weird on small viewports (because of the missing data attributes) ; please align with Antonio regarding this issue
- The paragraphs and headings are sensitive to colour changes (`#test-fake-global-rules`)

**Comment by emeryro** [comment]
- for tables, I added a comment on the ticket
- for text color, yes that's how the paragraphs and headings are defined. That's the same for non ecl-editor ones. Enforcing the color here may cause issues (what if the content has to be written in white forinstance?)

**Review by yhuard — APPROVED** [review]
good enough for now, we'll tackle the existing issues later

**Pull request merged**